File: squarecity.py
X = int(input())

base = 1
mult = 0
for m in range(1, X, 4):
    mult = mult + 2
    base = mult + 1

if base + mult == X: print(mult * (X - 2))
elif base + mult == X + 2: print(mult * X)

File: Otimizados/squarecity.c
#include <stdio.h>

int main(void) {
    int X;
    scanf("%d", &X);

    int base, mult = 0, m;
    for (m = 1; m < X; m += 4)
        mult += 2;

    base = mult + 1;

    if (base + mult == X) printf("%d\n", mult * (X - 2));
    else if (base + mult == X + 2) printf("%d\n", mult * X);

    return 0;
}